The original regex for parsing Bison's version output somehow doesn't
work at all with recent Bison. This one works with Bison 2.3 and 3.0.4,
which covers our bases pretty well.

Index: configure
===================================================================
--- configure.orig
+++ configure
@@ -51,7 +51,7 @@ my $bison_version;
 my $bison_letter;
 my $bison_found = 0;
 while (<IN>) {
-  if (/[Bb]ison[^0-9]*(([0-9.]+)([a-z]+)?)*\s*$/) {
+  if (/^bison[^0-9]+(([0-9.]+)([a-z]*))/) {
     my $whole_version = $1;
     $bison_version = $2;
     $bison_letter = $3;
